src/main/java/Ethereum.java

```diff
@@ -0,0 +1,70 @@
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.security.SignatureException;
+import java.util.HexFormat;
+
+import org.web3j.crypto.ECKeyPair;
+import org.web3j.crypto.Sign;
+import org.web3j.crypto.StructuredDataEncoder;
+
+/**
+ * References:
+ * - https://eips.ethereum.org/EIPS/eip-712
+ * */
+public class Ethereum {
```

Comment:
I think rather than inline comments, which I'm fine leaving or not, I'd say javadoc each method with this information in aggregate might read better and certainly in tooling it would be better.